pkg/core/dao/dao.go

```diff
@@ -471,9 +472,19 @@ func (dao *Simple) DeleteStorageItem(scripthash util.Uint160, key []byte) error
 
 // GetStorageItems returns all storage items for a given scripthash.
 func (dao *Simple) GetStorageItems(hash util.Uint160) (map[string]*state.StorageItem, error) {
+	return dao.GetStorageItemsWithPrefix(hash, nil)
+}
+
+// GetStorageItemsWithPrefix returns all storage items with given prefix for a
+// given scripthash.
+func (dao *Simple) GetStorageItemsWithPrefix(hash util.Uint160, prefix []byte) (map[string]*state.StorageItem, error) {
```

Comment:
I think it is better not to return map in DAO as the returned order is unpredictable (remember #822).
And if we need random access, new method or a use of bigger prefix are OK IMO.

Reply:
Argh, I keep forgetting about this. But then we probably also don't need `string` conversion here.

Reply:
Well, that may be a separate task as our current `GetStorageItems` API does the same thing (and that's why `WithPrefix` inherited it).

Reply:
#909.